ui/selectors/multi-srp/multi-srp.ts

```diff
@@ -44,7 +45,10 @@ const isPrimaryHdOrFirstPartySnapAccount = createDeepEqualSelector(
       return true;
     }
 
-    if (isMultichainWalletSnap(account.metadata.snap?.id as SnapId)) {
+    if (
+      isMultichainWalletSnap(account.metadata.snap?.id as SnapId) &&
+      account.options?.entropySource === primaryKeyring.metadata.id
```

Comment:
Very nit: Could be nice to have this as a selector like: `isHdAccountFromPrimarySrp`? (just an idea, but not mandatory here)